samples/apps/copilot-chat-app/webapi/Controllers/ChatHistoryController.cs

```diff
@@ -0,0 +1,193 @@
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="chatSessionRepository">The chat session repository.</param>
+    /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="promptSettings">The prompt settings.</param>
+    public ChatHistoryController(
+        ILogger<SemanticKernelController> logger,
+        ChatSessionRepository chatSessionRepository,
+        ChatMessageRepository chatMessageRepository,
+        PromptSettings promptSettings)
+    {
+        this._logger = logger;
+        this._chatSessionRepository = chatSessionRepository;
+        this._chatMessageRepository = chatMessageRepository;
+        this._promptSettings = promptSettings;
+    }
+
+    /// <summary>
+    /// Create a new chat session and populate the session with the initial bot message.
+    /// </summary>
+    /// <param name="chatParameters">Object that contains the parameters to create a new chat.</param>
+    /// <returns>The HTTP action result.</returns>
+    [Authorize]
+    [HttpPost]
+    [Route("chat/create")]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> CreateChatAsync(
+        [FromBody] ChatSession chatParameters)
+    {
+        var userId = chatParameters.UserId;
+        var title = chatParameters.Title;
+
+        var newChat = new ChatSession(userId, title);
+        await this._chatSessionRepository.CreateAsync(newChat);
+
+        var initialBotMessage = this._promptSettings.InitialBotMessage;
+        await this.SaveResponseAsync(initialBotMessage, newChat.Id);
+
+        this._logger.LogDebug("Created chat session with id {0} for user {1}.", newChat.Id, userId);
+        return this.CreatedAtAction(nameof(this.GetChatByIdAsync), new { chatId = newChat.Id }, newChat);
+    }
+
+    /// <summary>
+    /// Get a chat session by id.
+    /// </summary>
+    /// <param name="chatId">The chat id.</param>
+    [Authorize]
+    [HttpGet]
+    [ActionName("GetChatByIdAsync")]
+    [Route("chat/getChat/{chatId:guid}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetChatByIdAsync(Guid chatId)
+    {
+        var chat = await this._chatSessionRepository.FindByIdAsync(chatId.ToString());
+        if (chat == null)
+        {
+            return this.NotFound($"Chat of id {chatId} not found.");
+        }
+
+        return this.Ok(chat);
+    }
+
+    /// <summary>
+    /// Get all chat sessions associated with a user. Return an empty list if no chats are found.
+    /// </summary>
+    /// <param name="userId">The user id.</param>
+    [Authorize]
+    [HttpGet]
+    [Route("chat/getAllChats/{userId:regex([[0-9]]+\\.[[0-9]]+)}")]
```

Comment:
if we want to make sure the user id is guid, use `{userId:guid}` should be enough.

Reply:
The user id is not a guid. It's actually two guid string separated by a period.

Reply:
uh. I see. thanks.